projects/plugins/jetpack/extensions/blocks/videopress/edit.native.js

```diff
@@ -0,0 +1,402 @@
+
+		return <Icon icon={ SvgIcon } { ...iconStyle } />;
+	}
+
+	getVideoURL() {
+		const { attributes } = this.props;
+		const { guid } = attributes;
+		const { metadata = {} } = this.state;
+
+		// Private videos will open in the player for logged in users with the WordPress.com URL.
+		// However, they'll still display blank in the editor.
+		// TODO: We need to iterate so that private videos display as expected.
+		return metadata.is_private
+			? `https://video.wordpress.com/v/${ guid }`
+			: `https://videopress.com/v/${ guid }`;
+	}
+
+	render() {
+		const { setAttributes, attributes, isSelected, wasBlockJustInserted } = this.props;
+		const { id, src, guid } = attributes;
+		const { isLoadingMetadata, isUploadInProgress, isUploadFailed, videoContainerHeight } =
+			this.state;
+
+		const toolbarEditButton = (
+			<MediaUpload
+				allowedTypes={ [ MEDIA_TYPE_VIDEO ] }
+				isReplacingMedia={ true }
+				onSelect={ this.onSelectMediaUploadOption }
+				onSelectURL={ this.onSelectURL }
+				render={ ( { open, getMediaOptions } ) => {
+					return (
+						<ToolbarGroup>
+							{ getMediaOptions() }
+							<ToolbarButton
+								label={ __( 'Edit video', 'jetpack' ) }
+								icon={ replace }
+								onClick={ open }
+							/>
+						</ToolbarGroup>
+					);
+				} }
+			></MediaUpload>
+		);
+
+		const isSourcePresent = src || ( guid && id );
+		if ( ! isSourcePresent ) {
+			return (
+				<View style={ { flex: 1 } }>
+					<MediaPlaceholder
+						allowedTypes={ [ MEDIA_TYPE_VIDEO ] }
+						onSelect={ this.onSelectMediaUploadOption }
+						onSelectURL={ this.onSelectURL }
+						icon={ this.getIcon( ICON_TYPE.PLACEHOLDER ) }
+						onFocus={ this.props.onFocus }
+						autoOpenMediaUpload={ isSelected && wasBlockJustInserted }
+					/>
+				</View>
+			);
+		}
+
+		return (
+			<TouchableWithoutFeedback
+				accessible={ ! isSelected }
+				onPress={ this.onVideoPressed }
+				disabled={ ! isSelected }
+			>
+				<View style={ { flex: 1 } }>
+					{ ! this.state.isCaptionSelected && <BlockControls>{ toolbarEditButton }</BlockControls> }
+					{ isSelected && (
+						<InspectorControls>
+							<PanelBody title={ __( 'Settings', 'jetpack' ) }>
+								<VideoCommonSettings setAttributes={ setAttributes } attributes={ attributes } />
+							</PanelBody>
+						</InspectorControls>
+					) }
+					<MediaUploadProgress
+						mediaId={ id }
+						onFinishMediaUploadWithSuccess={ this.finishMediaUploadWithSuccess }
+						onFinishMediaUploadWithFailure={ this.finishMediaUploadWithFailure }
+						onUpdateMediaProgress={ this.updateMediaProgress }
+						onMediaUploadStateReset={ this.mediaUploadStateReset }
+						renderContent={ ( { retryMessage } ) => {
+							const videoURL = this.getVideoURL();
+							const showVideo =
+								isURL( videoURL ) &&
+								! isUploadInProgress &&
+								! isUploadFailed &&
+								! isLoadingMetadata;
+							const icon = this.getIcon( isUploadFailed ? ICON_TYPE.RETRY : ICON_TYPE.UPLOAD );
```

Comment:
I'll make a note to test offline behavior here. Now that we have the [useNetworkConnectivity](https://github.com/WordPress/gutenberg/blob/0847505149bc48eab957343323b429aba114fa85/packages/compose/src/hooks/use-network-connectivity/index.native.js) hook, there are ongoing design discussions where we may want to consider using an offline icon when network connectivity is offline in a follow-up PR.

Reply:
Also noting that [useNetworkConnectivity](https://github.com/WordPress/gutenberg/blob/0847505149bc48eab957343323b429aba114fa85/packages/compose/src/hooks/use-network-connectivity/index.native.js) is in the Gutenberg repo, so matching offline VideoPress behavior with other Media components that live in Gutenberg may be a bit of a bigger undertaking. 😅